docs/source/download.md

```diff
@@ -0,0 +1,41 @@
+# Download
+
+The official Apache DataFusion releases are provided as source artifacts.
+
+## Releases
+
+The latest source release is [0.45.0](https://www.apache.org/dyn/closer.lua/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz?action=download) ([asc](https://downloads.apache.org/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz.asc),
```

Comment:
Once we make the first release, we can update these links and merge this patch.

Reply:
<img width="1728" alt="image" src="https://github.com/apache/datafusion/assets/18818196/8c323379-e2da-40ac-8f8d-3d6abce636a5">